File: src/main/java/by/kohanova/model/Friends.java
package by.kohanova.model;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "friends")
public class Friends implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Integer id;

    @ManyToOne
    @JoinColumn(name = "friend1_id")
    public User user1;

    @ManyToOne
    @JoinColumn(name = "friend2_id")
    public User user2;

    @Column
    public boolean status;

    @Override
    public boolean equals(Object obj) {
        if (obj == null)
            return false;
        if (!(obj instanceof Friends))
            return false;
        Friends other = (Friends) obj;
        return id == null ? false : id.equals(other.id);
    }

    @Override
    public int hashCode() {
        return id == null ? 0 : id.hashCode();
    }
}

File: client/src/app/model/friends.ts
import {User} from "./user";
interface IFriends {
    id?: number;
    friend1_id?: User;
    friend2_id?: User;
    status?: boolean;
}
export class Friends implements IFriends {
    id: number;
    friend1_id: User;
    friend2_id: User;
    status: boolean;

    constructor(obj: IFriends = {}as IFriends) {
        let {
            friend1_id = null,
            friend2_id = null,
            status = false,

        }:IFriends = obj as IFriends;
        this.friend1_id = friend1_id;
        this.friend2_id = friend2_id;
        this.status = status;

    }
}